chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
Yes, currently `clock` and `reset` are available in user-level Module code. This PR does nothing to change that, however, your example illustrates the problem. Inside of a Module, `clock` refers to the implicit Clock input port to the Module. Thus, in your example, `subModule.clock` gets connected to the `clock` from the Module, **not** `io.otherClock`